dashboard/src/actions/index.js

```diff
@@ -197,3 +198,196 @@ export const addScratchpadEntry = entry => ({
+        _uid_
+    }
+}`;
+
+    timeout(
+        6000,
+        fetch(dgraphQuery(false), {
+            method: "POST",
+            mode: "cors",
+            headers: {
+                Accept: "application/json",
+                "Content-Type": "text/plain"
+            },
+            body: checkQuery
+        })
+            .then(checkStatus)
+            .then(response => response.json())
+            .then(function handleResponse(result) {
+                if (result.query && result.query.length > 0) {
+                    dispatch(updateShareId(result.query[0]["_uid_"]));
+                } else {
+                    // Else do a mutation to store the query.
+                    doShareMutation(dispatch, getState);
+                }
+            })
+    )
+        .catch(function(error) {
+            console.log(error.stack);
+            return error.message;
+        })
+        .then(function(errorMsg) {
+            if (errorMsg !== undefined) {
+                dispatch(
+                    saveErrorResponse(
+                        "Got error while saving querying for share: " + errorMsg
+                    )
+                );
+            }
+        });
+};
+
+export const getQuery = shareId => {
+    return dispatch => {
+        timeout(
+            6000,
+            fetch(dgraphQuery(false), {
+                method: "POST",
+                mode: "cors",
+                headers: {
+                    Accept: "application/json"
+                },
+                body: `{
+                    query(id: ${shareId}) {
+                        _share_
+                    }
+                }`
+            })
+                .then(checkStatus)
+                .then(response => response.json())
+                .then(function handleResponse(result) {
+                    if (result.query && result.query.length > 0) {
+                        dispatch(
+                            selectQuery(decodeURI(result.query[0]["_share_"]))
+                        );
+                        return;
+                    }
+                    dispatch(queryFound(false));
+                })
+        )
+            .catch(function(error) {
+                console.log(error.stack);
+                return error.message;
+            })
+            .then(function(errorMsg) {
+                if (errorMsg !== undefined) {
+                    dispatch(
+                        saveErrorResponse(
+                            `Got error while getting query for id: ${shareId}, err: ` +
+                                errorMsg
+                        )
+                    );
+                }
+            });
+    };
+};
+
+const updateAllowed = allowed => ({
+    type: "UPDATE_ALLOWED",
+    allowed: allowed
+});
+
+export const initialServerState = () => {
+    return dispatch => {
+        timeout(
+            6000,
+            fetch([dgraphAddress(), "ui/init"].join("/"), {
```

Comment:
Could we declare the endpoint as a variable for readability?

    const endpoint = [dgraphAddress(), "ui/init"].join("/")